tests/test_external_mixin.py

```diff
@@ -395,3 +401,141 @@ def test_restore_state(self, strategy):
         assert strategy._connected is False
         assert strategy._use_template_strategy is False
         strategy.state.restore_state.assert_called_once_with(state_dict)
+
+    @parameterized.expand([
+        [ext_strategy_fixture(LoadForecastExternalStrategy())],
+        [ext_strategy_fixture(PVForecastExternalStrategy())]
+    ])
+    def test_set_energy_forecast(self, strategy):
+        transaction_id = str(uuid.uuid4())
+        strategy._get_transaction_id = Mock(return_value=transaction_id)
+        arguments = {"transaction_id": transaction_id,
+                     "energy_forecast": {now().format(DATE_TIME_FORMAT): 1}}
+        payload = {"data": json.dumps(arguments)}
+        assert strategy.pending_requests == deque([])
+        strategy._set_energy_forecast(payload)
+        strategy._get_transaction_id.assert_called_with(payload)
+        assert len(strategy.pending_requests) > 0
+        energy_forecast_response_channel = f"{strategy.channel_prefix}/" \
+                                           "response/set_energy_forecast"
+        assert (strategy.pending_requests ==
+                deque([IncomingRequest("set_energy_forecast", arguments,
+                                       energy_forecast_response_channel)]))
+
+        strategy.redis.publish_json = Mock()
```

Comment:
Here you start a different set of tests. If you don't want to put them into a separate method, it would be great if you could add a comment to state how they differ from the previous ones from the lines above (i.e. the payload is now empty).